### PR TITLE
Revert "Remove curl after the usage (#6347)"

### LINKED ISSUE #6365: "Unhealthy" status reported by Docker because curl was removed
#6347 removes curl, however curl is still required by minio/dockerscripts/healthcheck.sh

## Expected Behavior
Docker should report correct heath status of Minio (i.e. running  `/usr/bin/healthcheck.sh` should only raise an error when the healthcheck has failed)

## Current Behavior
Docker reports that Minio is unhealthy.

## Possible Solution
1. Add curl back (using v7.61.0-r0 or greater) (removed by #6347 due to #6343)
2. Use an alternative to curl
3. Remove the heathcheck feature from Docker build

## Steps to Reproduce (for bugs)

1.  Deploy Minio using Docker per instructions at https://docs.minio.io/docs/minio-docker-quickstart-guide.html
2. Wait at least 150 seconds

## Context
Trying to deploy Minio with Docker...just a basic setup. Nothing fancy.

## Regression
Yes: a73b65de08d61e817f6507d3d16b9ba70394c43a 

## Your Environment

* Version used (`minio version`): 2018-08-25T01:56:38Z
* Environment name and version (e.g. nginx 1.9.1): Docker 18.06.1-ce
* Server type and version: N/A
* Operating System and version (`uname -a`): N/A
* Link to your project: N/A

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
This reverts commit dafa5073cb3dce6be039f5b27733ce71ab1b8fea.

## Motivation and Context
This commit revert also fixes #6365

## Regression
Yes, dafa5073cb3dce6be039f5b27733ce71ab1b8fea caused regression.

## How Has This Been Tested?
Manually using docker compose

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

File: Dockerfile [hunk 1]
```diff
@@ -14,7 +14,7 @@ WORKDIR /go/src/github.com/minio/
 COPY dockerscripts/docker-entrypoint.sh dockerscripts/healthcheck.sh /usr/bin/
 
 RUN  \
-     apk add --no-cache ca-certificates && \
+     apk add --no-cache ca-certificates curl && \
```
Comment (nitisht) [inline Dockerfile]: While we're at it, can we update Dockerfile to pick up the curl version 7.61.0-r0 or above as @sanverm2 suggested https://github.com/minio/minio/issues/6343#issue-353237646

Comment (harshavardhana) [reply Dockerfile]: It picks up automatically the fixed version @nitisht we don't need to specifically stick any version. But we can perhaps add > 7.61 deps

**Review by nitisht — APPROVED** [review]

**Review by krisis — APPROVED** [review]

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

**Pull request merged**